[trustar-150] Trustar v2 enhancements

hernantain opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21029

## Description

Multiple optional parameters have been added to `trustar-search-indicators` and `trustar-search-reports` commands. This lets the user search indicators/reports using more filters.

New optional argument `redact` on `trustar-submit-report` that gives users the possibility to redact a report before submission or not. By default the argument takes a 'NO' value (having the same functionality as before) but the user can specify `redact=YES` to redact the report (applying all terms in user company's redaction library to body and title) prior to submission.   

`trustar-get-phishing-submissions` and `trustar-get-phishing-indicators` were not working due to changes on TruSTAR's API. Phishing commands have been fixed and optional parameter `limit` has been added. 


## Screenshots
New parameters on `trustar-search-indicators` (similar for `trustar-search-reports`)

![335B9420-7A05-451D-A3E5-8C66CA3DB46B_1_105_c](https://user-images.githubusercontent.com/18565325/90172380-51b7c200-dd79-11ea-99e0-40e8e24095f6.jpeg)

`trustar-redact-report` command
![AF643086-3516-4D8F-BDE6-048C1B73BC0F_1_105_c](https://user-images.githubusercontent.com/18565325/90172497-7a3fbc00-dd79-11ea-9254-b5aa90586fb3.jpeg)


## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [x] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
```diff
@@ -745,18 +762,78 @@ script:
     outputs:
-    - contextPath: TruSTAR.Report.id
-      description: ID of the report
+    - contextPath: TruSTAR.Indicators.indicatorType
```
This contexts shouldn't have changed. I already changed them back as they were before. Thanks for noticing.

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
Same with previous comment. This shouldn't have changed.

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.yml:
```diff
@@ -936,12 +1019,51 @@ script:
+      predefined:
+      - COMMUNITY
+      - ENCLAVE
+      description: Distribution type of the report
```
Done

hernantain commented on Packs/TruSTAR/ReleaseNotes/2_0_1.md:
```diff
@@ -0,0 +1,10 @@
+#### Integrations
+##### TruSTAR v2
+
+Multiple optional parameters have been added to `trustar-search-indicators` and `trustar-search-reports` commands. This lets the user search indicators/reports using more filters.
```
Done

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.py:
```diff
@@ -632,6 +660,44 @@ def search_reports(self, search_term=None, enclave_ids=None):
+
+        :return: Entry context with the submitted report.
+        """
+        response = self.client.redact_report(
```
There are 2 different functionalities here. In some cases we want to let the user submit a report with the literal title and body and in others they can 'redact' a report. This means that some terms that they have on their redaction library will be applied, so those terms on the redaction library will be replaced by a generic tag. In this particular command when a user redacts a report, it will also be submitted with the redaction terms applied.

You can see an example on our API docs here: https://docs.trustar.co/api/v13/reports/redact_report.html

hernantain commented on Packs/TruSTAR/Integrations/TruSTAR_V2/TruSTAR_V2.py:
```diff
@@ -632,6 +660,44 @@ def search_reports(self, search_term=None, enclave_ids=None):
+        redacted_title = response.title
+        redacted_body = response.body
+
+        return self.submit_report(
```
Done

hernantain commented:
@barchen1 Comments Addressed!

barchen1 commented:
@hernantain Thanks for the Demo
Here are the things we need to test/add before merging this PR:
1. Have a better error handling for wrong api key/api secret.
2. Add a description for "enclave ids" argument which explain how the user can get his enclave id (using get-enclaves command)
3. Add a description/example for the arguments that can be comma-separated like encalve_ids, tags..
4. Test the "search-reports" command with from_time and to_time arguments.
5. Test the "search-reports" command with excluded_tags argument.
6. Add a description/example for from_time and to_time arguments so the user will know he can enter "20 minutes ago" and not just a timestamp. 
7. Test the search-indicators with tags argument and return a better error message or an empty list if there aren't any indicators with the given tag. 
8. We recommend to edit the test playbook to include the enhancements of the commands.

Let me know if you have any questions.
Thanks

hernantain commented:
@barchen1 I implemented the changes. Improved descriptions and examples as mentioned, fixed error handling for creds issues and search-indicators command, and tested/fixed the search-reports command

barchen1 commented on Packs/TruSTAR/ReleaseNotes/2_0_1.md:
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+##### TruSTAR v2
+  - Arguments added
+    - Multiple optional arguments have been added to `trustar-search-indicators` and `trustar-search-reports` commands. This lets the user search indicators/reports using more filters.
+    - New optional argument `redact` on `trustar-submit-report` that gives users the possibility to redact a report before submission or not. By default the argument takes a 'NO' value (having the same functionality as before) but the user can specify `redact=YES` to redact the report (applying all terms in user company's redaction library to body and title) prior to submission.   
+    
+  - Fixes
+    - `trustar-get-phishing-submissions` and `trustar-get-phishing-indicators` were not working due to changes on TruSTAR's API. Phishing commands have been fixed and optional parameter `limit` has been added. 
+
```
```suggestion
#### Integrations
##### TruSTAR v2
Added support for multiple new filtering arguments to the **trustar-search-indicators** and the **trustar-search-reports** commands.
Added optional argument *redact* to the **trustar-submit-report** command which gives users the possibility to redact a report before submission.
Fixed an issue where the  **trustar-get-phishing-submissions** and  the **trustar-get-phishing-indicators** did not work due to changes in the TruSTAR API.
Added support for *limit* argument in the **trustar-get-phishing-submissions** and  the **trustar-get-phishing-indicators**.
```

barchen1 reviewed (changes requested):
Did you test the search-reports command ? 
cause the search_reports function stayed the same.

hernantain commented:
@barchen1 Yes. The issue we had on the demo was because I was using the test integration were I was missing some lines of code on that command. That's why the `from_time` and `to_time` parameters were failing. Then I also tested the filtering using tags and excluded_tags and is working fine.
